packages/models-library/src/models_library/folders.py

```diff
@@ -1,13 +1,41 @@
 from datetime import datetime
+from enum import auto
 from typing import TypeAlias
 
-from models_library.users import GroupID, UserID
-from models_library.workspaces import WorkspaceID
-from pydantic import BaseModel, Field, PositiveInt
+from pydantic import BaseModel, Field, PositiveInt, validator
+
+from .users import GroupID, UserID
+from .utils.enums import StrAutoEnum
+from .workspaces import WorkspaceID
 
 FolderID: TypeAlias = PositiveInt
 
 
+class FolderScope(StrAutoEnum):
+    ROOT = auto()
+    SPECIFIC = auto()
```

Comment:
While I get the other two options I have no idea what SPECIFIC means in this context. Could you please explain this?

Reply:
The idea is to differentiate the type of filtering to apply. For folders, this could mean:

- Root folder (represented by a null value in the database)
- Specific folder (represented by an integer)
- ALL folders (searching through all folders)

The same logic applies to workspaces. In the context of workspaces, the types are:

- PRIVATE (My Workspace - represented by a null value in the database)
- SHARED (represented by an integer)
- ALL workspaces

I will add some comments in the code, to make it more clean.